### DISCUSSION TITLE
Transpile PHP (Phalcon Micro) to Swoole or DrogonCore (HTTP Asynchronous)

### DISCUSSION THREAD

**Discussion by fakharak** [phalcon/cphalcon#15346] [General]
On the Techempower benchmark below, DrogonCore (C++ 14/17 based Micro & Asynchronous HTTP Framework) is on top in throughput as it is able to utilize 100% computing-resources, efficiently.
https://www.techempower.com/benchmarks/

Can you please transpile PHP code (like, from within in Phalcon Micro) to DrogonCore ? 
Link:
https://github.com/an-tao/drogon

Even better if you can import Swoole (in the form of C Library libswoole) to "Phalcon Micro" then Phalon Micro will make the best selling point. Swoole is also available in the form of C Library. Swoole is a framework for Asynch Network / IO Programming which supports multiple protocols
Useful links:
https://wiki.swoole.com/wiki/page/633.html
https://github.com/swoole/swoole-src
https://github.com/guweigang/swoole

I have done bridging of Phalcon Micro and Swoole on PHP level but as Phalcon Micro is supported upto 7.4 (not on PHP 8.0 so far) so it will be great if Phalcon 4.2 imports / bridges Swoole to add it with its compiled- Micro/MVC/HMVC.

Team behind Swoole is available on slack and twitter, let me know if you need links. They have same mission as yours which is to improve PHP Performance, and add concurrency paradigm.

I will try to come up to your level in C++ but can not promise now, however as you guys are already expert and have a hands on with C++, so perhaps such bridging will be relatievly easier for you, i hope.

**Comment by Jeckerson** [comment]
Hello

> I have done bridging of Phalcon Micro and Swoole on PHP level but as Phalcon Micro

Could you open source it and share?


> Team behind Swoole is available on slack and twitter, let me know if you need links. They have same mission as yours which is to improve PHP Performance, and add concurrency paradigm.

Phalcon isn't fully PSR compliant with v4, but there are plans after v5 is released. With that is will be easier to bridge whatever is also psr compliant.

> I will try to come up to your level in C++ but can not promise now, however as you guys are already expert and have a hands on with C++, so perhaps such bridging will be relatievly easier for you, i hope.

Zephir isn't written in C++, but in C. And for PHP8 support you can track progress here: https://github.com/zephir-lang/zephir/issues/2111

**Comment by fakharak** [comment]
- Bridging of (compiled) "Phalcon Micro" with (compiled) "Swoole" (on PHP code level).
Repo:
https://github.com/fakharak/CrustPHP

- It is already open-sourced.
- The code above is well-tested with PHP 7.4.
- User Authentication Logic is currently stuffed into Controller because providing full MVC structure is beyond intent; i have added Authentication/Controller merely for providing Proof of Concept. (I will provide you User Table, if you will need)

**Comment by fakharak** [comment]
@Jeckerson 

Laravel Octane has provided its integration with Swoole, just like i bridged Phalcon Micro and Swoole, in my repo CrustPHP.
Mezzio (old name Zend Expressive) from Linux Foundation had already done integration with Swoole.

May be Phlcon (or, Phalcon Micro) can do better with PHP8.0 by porting Swoole (Swoole is ready for Asynchronous PHP8.0 and Swoole is also available for C-language as libswoole) ?  and / or by porting DrogonCore ?

**Reply by Jeckerson** [reply]
There is internal work in community to integrate Phalcon with Swoole.

For further details ask @kaioken (in [Discord](http://phalcon.io/discord)).
